Core concepts plus styling

JAndresHJ opened the pull request.

JAndresHJ committed:
CHALLENGE_1 - Renders the header and the list of videos

JAndresHJ committed:
CHALLENGE_1 - Unit tests

AlejandroSeguraWIZ commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,157 @@
+    >
+      <MenuItem onClick={handleMenuClose}>Login</MenuItem>
+    </Menu>
+  );
```
This could be moved to another component like `DesktopMenu`

AlejandroSeguraWIZ commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,157 @@
+        <p>Profile</p>
+      </MenuItem>
+    </Menu>
+  );
```
This could be moved to another component like `MobileMenu`

AlejandroSeguraWIZ commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,157 @@
+        </Toolbar>
+      </AppBar>
+      {renderMobileMenu}
+      {renderMenu}
```
This could be replaced by:
```
<DesktopMenu {...someProps} />
<MobileMenu {...otherProps} />
```

AlejandroSeguraWIZ commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,157 @@
+            >
+              <AccountCircle />
+            </IconButton>
+          </div>
```
This could be another component like `DesktopActionsHeader`

AlejandroSeguraWIZ commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,157 @@
+            >
+              <MoreIcon />
+            </IconButton>
+          </div>
```
This could be another component like `MobileActionsHeader`

AlejandroSeguraWIZ commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,157 @@
+    </Menu>
+  );
+
+  return (
```
I think this component is way too long. It should be split into different pieces to make them more granular, readable, and maintainable.
A possibility could be turning the component into the next structure:
```
<Header>
     <BurgerMenu />
     <SearchInput />
     <DesktopActions />
     <MobileActions />
</Header>
```

AlejandroSeguraWIZ commented on src/components/Videos/Videos.js:
```diff
@@ -0,0 +1,25 @@
+// Utils
+import mockedVideos from '../../utils/youtube-videos-mock.json';
+
+const Videos = () => {
```
Probably a better name for this component could be `VideoList`

AlejandroSeguraWIZ reviewed (changes requested):
I do like your project. The structure is awesome. I just left some comments regarding the separation of concerns for each component, but overall good job!

Also, I recommend you to put the test files in a folder `__tests__`. i.e.
```
Header/
    __tests__/
         Header.test.js
```

JAndresHJ committed:
Addresses comments

JAndresHJ commented:
Hi @AlejandroSeguraWIZ many thanks for the review! :smiley: I have addressed your comments, you were right, that header component definitely needed to be splitted.
